util/src/main/java/io/kubernetes/client/informer/cache/SharedProcessor.java

```diff
@@ -146,7 +147,17 @@ public void stop() {
     } finally {
       lock.writeLock().unlock();
     }
-    // TODO(yue9944882): gracefully shutdown listener pool
-    executorService.shutdownNow();
+    // Disable new tasks from being submitted
+    executorService.shutdown();
+    try {
+      // Wait a while for existing tasks to terminate
+      if (!executorService.awaitTermination(1, TimeUnit.MINUTES)) {
+        // Cancel currently executing tasks
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      executorService.shutdownNow();
+      Thread.currentThread().interrupt();
```

Comment:
can you elaborate the point of this line? interrupting the current thread doesnt make sense to me

Reply:
Yes, there is no need to interrupt the thread here. (There is no higher level interrupt handlers so it doesn't need to reraise the InterruptedException)

I have removed it.